lib/sidekiq/instrument/middleware/server.rb

```diff
@@ -17,15 +17,21 @@ def call(worker, job, _queue, &block)
       execution_time_ms = (Time.now - start_time) * 1000
       Statter.statsd.measure(metric_name(worker, 'runtime'), execution_time_ms)
       Statter.dogstatsd&.timing('sidekiq.runtime', execution_time_ms, worker_dog_options(worker))
-    rescue StandardError => e
+    rescue Exception => e
```

Comment:
This will notably catch signals, which might produce weird results during shutdown. If there's some particular error you care about that's not in the `StandardError` family tree you might want to just specifically rescue that.

Reply:
:this:

Reply:
We're only rescuing in order to increment metrics, then simply re-raising the error a few lines later. I understand you shouldn't normally rescue the whole Exception umbrella, but this is a particularly unique case. If we don't rescue the entire Exception umbrella, we don't increment our stats in certain error cases and it makes us have incorrect enqueue vs dequeue vs error counts.

Reply:
What error is this that's not based on `StandardError`?